main.go

```diff
@@ -25,30 +90,169 @@ func getNewsContent(url string) bytes.Buffer {
+	if response.StatusCode != http.StatusNoContent {
+		log.Printf("HTTP POST to Discord server failed. Status Code: %v", response.StatusCode)
+		body, readErr := ioutil.ReadAll(response.Body)
+		if readErr != nil {
+			log.Fatalf("Could not read response body. Error: %v", readErr)
+		}
+		log.Fatal(string(body))
+	} else {
+		fmt.Println("Success!")
+	}
+}
+
+func getSteamNews(appid int) newsResponse {
+	// get the latest news post and parse the json
+	url := fmt.Sprintf("https://api.steampowered.com/ISteamNews/GetNewsForApp/v2/?appid=%v&count=1", appid)
+	data := getAPIContent(url)
+	var steamResponse newsResponse
+	jsonErr := json.Unmarshal(data, &steamResponse)
+	if jsonErr != nil {
+		log.Fatalf("Could not process API response. Error: %v", jsonErr)
+	}
+	return steamResponse
+}
+
+func processNewsResponse(gidMap map[string]string, steamResponse newsResponse) {
+	// if the map is empty (due to starting up for the first time), populate it with the local file's GIDs
+	if len(gidMap) < 0 {
+		savedGids := readNewsGid()
+		for _, gid := range savedGids {
+			gidMap[string(gid)] = ""
+		}
+	}
+
+	// check if each news GID is in the map
+	// if not, add it and save to file in case the service dies for some reason
+	for _, item := range steamResponse.AppNews.NewsItems {
+
+		// primarily for startup, check if new posts within 1 hour so you dont
+		// spam the discord channel with the last news post. unless this is offline for several days, they're old anyway
+		if checkIfDateWithinHour(item.Date) {
+			// check if the gid's are loading into memory already
+			if _, ok := gidMap[item.Gid]; !ok {
+				gidMap[item.Gid] = ""
+				saveNewsGid(item.Gid)
+				// get game name, format message, send to discord
+				nameBytes := getAPIContent("https://api.steampowered.com/ISteamApps/GetAppList/v2/")
+				name := getGameName(steamResponse.AppNews.AppID, nameBytes)
+				fmt.Println(name)
+				postToDiscord(formatNewsMessage(steamResponse, name))
+			} else {
+				log.Println("Nothing new found")
+
+			}
+		} else {
+			log.Println("Nothing new found in last hour")
 		}
-		fmt.Println("Successfully loaded the latest news article into into memory!\n", url)
-		return buffer
 	}
-	return bytes.Buffer{}
 }
 
 func main() {
-	url := "https://api.steampowered.com/ISteamNews/GetNewsForApp/v2/?appid=717790&count=1"
-	data := getNewsContent(url)
-	fmt.Println(data.String())
+	// ensure logfile exists, if not, create it.
+	logfile := "news_updater.log"
+	_, statErr := os.Stat(logfile)
+	if os.IsNotExist(statErr) {
+		file, crErr := os.Create(logfile)
+		if crErr != nil {
+			fmt.Printf("Could not create logfile. Error: %v", crErr)
+		}
+		_ = file.Close()
+	}
+
+	// check for a new steam news post for a list of appids
+	gidMap := make(map[string]string)
+	for {
+		appIDs := []int{717790, 383120, 530870, 271590, 674370, 552990, 587120, 613100, 943130, 771800}
+		for _, appid := range appIDs {
+			newsJson := getSteamNews(appid) // use a new goroutine for steam news
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [var newsJson should be newsJSON](https://app.codacy.com/app/matthew.clairmont1/steam-news-alerter/pullRequest?prid=4554288)